telosnetwork/open-block-explorer
# Better filters for transaction table

# Fixes #547, #570, #549, #588
## Description 

Previously, the transaction table filters were not functioning as expected. To improve the user experience, more intuitive filter components were implemented, allowing users to easily understand and utilize the filters. Additionally, instead of solely filtering the current results, a new query is now initiated with the updated filter parameters.

Another issue that was addressed was the pagination buttons at the bottom of the table scrolling horizontally off the screen when the table width exceeded the viewport. This issue has been resolved to ensure a seamless user experience.

## Test scenarios

**NOTE**: this only works on **[Mainnet](https://547-scoped-search-should-con.open-block-explorer.pages.dev/)** because we don't have enough tokens (actually zero) and Hyperion-dev does not use the query param _notified_ so the account filter does not wor on Testnet.

Scenario: Filter by Account
- Click on the "Accounts" button
- Partially type a name and select an account from the dropdown list
- The button should change to a secondary color (blue), the button text should include the name of the selected account, and a new page should load displaying only the transactions associated with that account.

Scenario: Filter by Action Name
- Click on the "Actions" button
- Type one or more action names separated by commas
- The button should change to a secondary color (blue), the button text should include the typed action name(s), and a new page should load displaying only the transactions associated with the selected action(s).

Scenario: Filter by Date Range
- Click on the "Date" button
- Modify one or both of the selected dates
- Once the changes are complete, the button should change to a secondary color (blue), the button text should include the selected date range, and a new page should load displaying only the transactions within the selected date range.

Scenario: Filter by Tokens
- Click on the "Token" button
- Type one or two letters of the token symbol
- A dropdown list of tokens should appear
- Select one token from the list
- Once the selection is made, the button should change to a secondary color (blue), the button text should include the selected token symbol, and a new page should load displaying only the transactions associated with the selected token.

Scenario: Multiple Simultaneous Filters
- Any combination of the above scenarios should result in a group of secondary-colored buttons changing their text to reflect the applied filters. The transaction page should respect all filters at once, possibly resulting in zero results being displayed.

Scenario: Clear Filters
- Execute any of the above scenarios
- Note that a "Clear filters" button appears, containing a cross symbol and text
- Click on this button
- All filter buttons should return to their primary color (violet), and their text should return to their default label. The transaction page should contain an unfiltered list of transactions.

## Screenshots

![image](https://user-images.githubusercontent.com/4420760/218556823-b1f1578d-44a5-42a7-93ee-982d4c8893de.png)
![image](https://user-images.githubusercontent.com/4420760/218556859-412a32c8-79cf-4df5-a318-dac4673c74cd.png)
![image](https://user-images.githubusercontent.com/4420760/218556887-39c12c49-bbfa-4ea0-af71-ddfc91af2e75.png)
![image](https://user-images.githubusercontent.com/4420760/218557435-319d019c-02de-4c53-acd8-fd4ff157e95f.png)


## Checklist:
